client/src/components/GalleryItem/GalleryItem.js

```diff
@@ -38,35 +38,121 @@ const preventFocus = (event) => {
   event.preventDefault();
 };
 
-class GalleryItem extends Component {
-  constructor(props) {
-    super(props);
+const GalleryItem = ({
+  sectionConfig = {
+    imageRetry: {},
+  },
+  item = {},
+  loadState,
+  bustCache = true,
+  highlighted,
+  selected,
+  isDropping,
+  isDragging,
+  maxSelected,
+  message,
+  selectable,
+  onActivate,
+  onSelect,
+  onCancelUpload,
+  onRemoveErroredUpload,
+  badge,
+  updateStatusFlags = (flags) => flags,
+  updateProgressBar = (progressBar) => progressBar,
+  updateErrorMessage = (msg) => msg,
+  children,
+  actions,
+}) => {
+  const thumbnailRef = useRef(null);
+  const titleRef = useRef(null);
+
+  useEffect(() => {
+    if (shouldLoadImage({
+      item,
+      sectionConfig,
+    })) {
+      actions.imageLoad.loadImage(
+        item.thumbnail,
+        sectionConfig.imageRetry
+      );
+    }
+  }, [item, sectionConfig, actions]);
```

Comment:
```suggestion
  }, [item, sectionConfig, actions.imageLoad]);
```
We don't need to fire this effect if unrelated actions change.